src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/DirectoryCatalog.cs

```diff
@@ -664,12 +664,22 @@ public override string ToString()
         private AssemblyCatalog CreateAssemblyCatalogGuarded(string assemblyFilePath)
         {
             Exception exception = null;
-
             try
             {
-                return (_reflectionContext != null)
-                    ? new AssemblyCatalog(assemblyFilePath, _reflectionContext, this)
-                    : new AssemblyCatalog(assemblyFilePath, this);
+                try
+                {
+                    return (_reflectionContext != null)
+                        ? new AssemblyCatalog(assemblyFilePath, _reflectionContext, this)
+                        : new AssemblyCatalog(assemblyFilePath, this);
+                }
+                //fallback attempt issue https://github.com/dotnet/corefx/issues/27433
+                catch (FileNotFoundException)
+                {
+                    var loadedAssembly = Assembly.LoadFrom(assemblyFilePath);
```

Comment:
While this may address this specific scenario in DirectoryCatalog I think it is better to address it in more scenarios in AssemblyCatalog itself. I suggest adding the logic around https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/AssemblyCatalog.cs#L553.

Reply:
good point

Reply:
@weshaggard done